src/main/java/org/spongepowered/api/world/gen/WorldGenerator.java

```diff
@@ -35,16 +32,17 @@
 public interface WorldGenerator {
 
     /**
-     * Generates the chunk at the given position in the world. The position
-     * specified is the chunk position (ie. the world position divided by the
-     * chunk size). The newly generated chunk should be placed into the given
-     * {@link BlockBuffer}.
-     *
-     * @param world The world
-     * @param buffer The buffer to generate the region into
-     * @param position The chunk position
+     * Gets the {@link GeneratorPopulator} of this world generator.
+     * @return The {@link GeneratorPopulator}.
+     * @see #setGeneratorPopulator(GeneratorPopulator)
+     */
+    GeneratorPopulator getGeneratorPopulator();
```

Comment:
I originally had a List similar to the populators, but that didn't work due to implementation difficulties: how to add a GeneratorPopulator to the standard ChunkProviderGenerate? How to add it to an implemention of IChunkProvider created by some random mod? The first case can probably be handled using a mixin, but the second would require something more creative.

For normal populators we don't have this problem: we can just listen to an event Forge (and any well-behaving mod) fires.